### DISCUSSION TITLE
Problem animating entities

### DISCUSSION THREAD

**Discussion by victorbrndls** [google/filament#7070] [Q&A]
**Describe the bug**
I need to animate entities using Filament and the animation is not working correctly. I'm not sure if there's a bug or if I'm doing this wrong.

**To Reproduce**
Run the app at https://github.com/victorbrndls/FilamentSampleApp

This is the animation code:
```
entitiesToAnimate.forEach { entity ->
	val initialTransform = FloatArray(16)
	tcm.getTransform(entity, initialTransform)

	val animationMatrix = FloatArray(16)

	val rotateY = if (forward) -1f else 1f

	animator.addUpdateListener {
		val angle = it.animatedValue as Float

		System.arraycopy(initialTransform, 0, animationMatrix, 0, 16)

		Matrix.translateM(
			animationMatrix,
			0,
			-initialTransform[12],
			-initialTransform[13],
			-initialTransform[14],
		)
		Matrix.rotateM(animationMatrix, 0, angle, 0f, rotateY, 0f)
		Matrix.translateM(
			animationMatrix,
			0,
			initialTransform[12],
			initialTransform[13],
			initialTransform[14],
		)

		tcm.setTransform(entity, animationMatrix)
	}
}
```
I'm moving the pieces to the center, rotating them and moving them back.

**Expected behavior**
It should animate correctly

**Screenshots**
![1692018077405](https://github.com/google/filament/assets/11446530/54b464d1-2920-4abc-ad37-7309c6e34c17)
![1692018077431](https://github.com/google/filament/assets/11446530/ac5d02cc-6301-4e89-9d98-c749b08b026f)

**Smartphone (please complete the following information):**
 - Device: Redmi Note 7
 - OS: Android 10
Happen on any device

**Additional context**
Filament 1.40.5

**Comment by pixelflinger** [comment]
I think the discussions forum is more appropriate here.
